packages/tre/src/plugins/r2/gateway.ts

```diff
@@ -73,11 +130,33 @@ export interface R2Objects {
 
 const validate = new Validator();
 
+function generateVersion() {
+  return crypto.randomBytes(16).toString("hex");
```

Comment:
For future, if you want to move this code to a Worker then this API is not available and it is better to use the `crypto.webcrypto.getRandomValues()` api.